### ISSUE TITLE
Using `domain` in a formal tuple type issues error message

### ISSUE THREAD

**Issue by DanilaFe** [chapel-lang/chapel#23483]
In the following program:

```Chapel
proc f4(arg: (domain, int)) { }
```

The error message is quite poor:

```
domainTupFormal.chpl:1: syntax error: near ','
```

Likely, we don't even want this to be an error, since we'd rather accept the program, and warn that the `domain` formal type needs to have a `(?)`:

```Chapel
proc f4(arg: (domain(?), int)) { }
```

**Comment by bradcray** [comment]
Likely you're way ahead of me, but I'm wondering whether this is an offshoot of the fact that `TDOMAIN` is a special token in the lexer/parser, rather than taking the Andy approach of just treating it as a special identifier, and whether switching to the Andy approach would improve things here.  (That said, my real hope would be that we switch to the Daniel variation of the Andy approach in which the lexer still keeps it a reserved word, but then makes it part of the identifier rule in the parser so that nothing downstream is any the wiser).

**Comment by DanilaFe** [comment]
Yes, that's something like what happens. I think there's a good reason for it, but I haven't looked into it yet -- most likely, the issue is conflicts with `x.domain` and other "special" language forms.
